### PR TITLE
Feat: Adds tokens to `Alert`

### PR DESCRIPTION
**Pull request by renatamottam**
Cherry-pick from Gatsby Project: https://github.com/vtex-sites/gatsby.store/pull/77

## What's the purpose of this pull request?
This PR applies the new [Theming structure](https://github.com/vtex-sites/base.store/pull/407) to `Alert`: new scoped tokens linked to global ones.

## How does it work?

This PR uses local variables to parameterize the `Alert` properties, then connects these scoped tokens to the global ones.

### New local tokens for `Alert`

Screen Shot 2022-05-30 at 19 39 21

#### Default

|Local token|Default value / global token linked|
|-|-|
|`--fs-alert-height`                       | `var(--fs-spacing-7)`|
|`--fs-alert-padding-left`                 | `var(--fs-spacing-3)`|
|`--fs-alert-padding-right`                | `var(--fs-alert-padding-left)`|
|||
|`--fs-alert-bkg-color`                    | `var(--fs-color-highlighted-bkg)`|
|||
|`--fs-alert-text-color`                   | `var(--fs-color-highlighted-text)`|
|`--fs-alert-text-size`                    | `var(--fs-text-size-1)`|
|||
|`--fs-alert-link-color` | `var(--fs-alert-text-color)`|
|`--fs-alert-link-color-visited` | `var(--fs-alert-text-color)`|
|||
|`--fs-alert-button-text-color` | `var(--fs-alert-text-color)`|
|`--fs-alert-button-bkg-color` | `var(--fs-alert-bkg-color)`|
|`--fs-alert-button-border-radius` | `var(--fs-border-radius)`|

## How to test it?

The `Alert` component and its variants should look just as it was before these changes.
[Preview Link](https://nextjs-store-storybook-git-feat-theming-alert-cb9989-faststore.vercel.app/?path=/docs/molecules-alert--default-story)

## Checklist

- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 
- [x] Updated the Storybook - *if applicable*.
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.

### CODE REVIEW

File: src/components/ui/Alert/Alert.tsx [hunk 1]
```diff
@@ -7,9 +7,20 @@ import FrameworkLink from 'src/components/common/Link'
...
-type Props = AlertProps & {
+import styles from './alert.module.scss'
+
+export type Props = AlertProps & {
+  /**
+   * Icon component for additional customization
+   */
   icon?: ReactNode
+  /**
+   * Add the possibility to make the component dismissible
```
Comment (hellofanny) [inline src/components/ui/Alert/Alert.tsx:18]: `Enables dismissible feature` ?

File: src/components/ui/Alert/Alert.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      link: { text: 'Buy now', to: '/' },
+      dismissible: true,
+    }}
+  >
+    {Template.bind({})}
+  </Story>
+  <Story name="overview-clean">{Template.bind({})}</Story>
+  <Story
+    name="overview-big"
+    args={{
```
Comment (hellofanny) [inline src/components/ui/Alert/Alert.stories.mdx:53]: I don't like showing this possibility 😬 Shouldn't we avoid this case? I know we are handling this case and it's nice to show it. Maybe we could also add a recommendation not to use this component with such a long text, somewhere?!

Comment (hellofanny) [reply src/components/ui/Alert/Alert.stories.mdx:53]: Let's solve this in another PR!

**Review by hellofanny — APPROVED** [review]
Nicee! Let's add the do an don't sections from this component in another PR :)

File: src/components/ui/Alert/Alert.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      dismissible: true,
+    }}
+  >
+    {Template.bind({})}
+  </Story>
+</Canvas>
+
+<ArgsTable story="default" />
+
+<table
```
Comment (eduardoformiga) [inline src/components/ui/Alert/Alert.stories.mdx:84]: What do you think about using the new `TokenTable` format here?

**Review by eduardoformiga — APPROVED** [review]
Awesome!

I left a minor comment.

File: src/components/ui/Alert/Alert.stories.mdx (hunk 3 above)
Comment (renatamottam) [reply src/components/ui/Alert/Alert.stories.mdx:84]: I'll add it on another PR!

**Pull request merged**